backend/rust/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use backend::backend_server::{Backend, BackendServer};
+use backend::{HealthMessage, PredictOptions,Reply,ModelOptions, EmbeddingResult, GenerateImageRequest,TranscriptRequest,TranscriptResult,TtsRequest,TokenizationResponse};
+use tonic::{Request, Response, Status};
+use tokio_stream::{wrappers::ReceiverStream};
+
+use tonic::transport::Server;
+
+pub mod backend{
+    tonic::include_proto!("backend");
+}
+
+
+#[derive(Debug)]
+struct BackendService;
+
+#[tonic::async_trait]
+impl Backend for BackendService{
```

Comment:
Once the protobuf is stable probably you could publish a thin library crate with the pre-compiled backend.rs and the `Backend` trait + basic types.

Until then would be good to refactor `backend/rust` to be a workspace so the common grpc code can live in in a crate and the actual impl using burn, candle doesn't have to recompile the proto for each of them.

Reply:
Thank you. Good idea. Already changed to the workspace+ a trait package named `bunker`, burn package for implement it. And also added the Proto file `backends.rs` to the trait package.